backend-tests/tests/test_inventory.py

```diff
@@ -0,0 +1,243 @@
+    for _ in range(num_devices):
+        dev = make_accepted_device(utoken, devauthd, tenant_token=tenant_token)
+        devices.append(dev)
+
+    return devices
+
+class TestGetDevicesBase:
+    def do_test_get_devices_ok(self, user, tenant_token=''):
+        useradmm = ApiClient(useradm.URL_MGMT)
+        devauthd = ApiClient(deviceauth_v1.URL_DEVICES)
+        invm = ApiClient(inventory.URL_MGMT)
+        invd = ApiClient(inventory.URL_DEV)
+
+        # log in user
+        r = useradmm.call('POST',
+                          useradm.URL_LOGIN,
+                          auth=(user.name, user.pwd))
+        assert r.status_code == 200
+        utoken = r.text
+
+        # prepare accepted devices
+        devs = make_accepted_devices(utoken, devauthd, 40, tenant_token)
+
+        # wait for devices to be provisioned
+        time.sleep(3)
+
+        r = invm.with_auth(utoken).call('GET',
+                                        inventory.URL_DEVICES,
+                                        qs_params={'per_page':100})
+        assert r.status_code == 200
+        api_devs = r.json()
+        assert len(api_devs) == 40
+
+    def do_test_filter_devices_ok(self, user, tenant_token=''):
+        useradmm = ApiClient(useradm.URL_MGMT)
+        devauthd = ApiClient(deviceauth_v1.URL_DEVICES)
+        invm = ApiClient(inventory.URL_MGMT)
+        invd = ApiClient(inventory.URL_DEV)
+
+        # log in user
+        r = useradmm.call('POST',
+                          useradm.URL_LOGIN,
+                          auth=(user.name, user.pwd))
+        assert r.status_code == 200
+        utoken = r.text
+
+        # prepare accepted devices
+        devs = make_accepted_devices(utoken, devauthd, 40, tenant_token)
+
+        # wait for devices to be provisioned
+        time.sleep(3)
+
+        r = invm.with_auth(utoken).call('GET',
+                                        inventory.URL_DEVICES,
+                                        qs_params={'per_page':100})
+        assert r.status_code == 200
+        api_devs = r.json()
+        assert len(api_devs) == 40
+
+        # upload inventory attributes
+        for i, d in enumerate(devs):
+            payload = [
+                    {
+                        "name": "mac",
+                        "value": "de:ad:be:ef:06:" + str(i)
+                    }
+            ]
+            r = invd.with_auth(d.token).call('PATCH',
+                                        inventory.URL_DEVICE_ATTRIBUTES,
+                                        payload)
+            assert r.status_code == 200
+
+        # filter devices
+        qs_params={}
+        qs_params['per_page'] = 100
+        qs_params['mac'] = 'regex:06:2'
+        r = invm.with_auth(utoken).call('GET',
+                                        inventory.URL_DEVICES,
+                                        qs_params=qs_params)
+        assert r.status_code == 200
+        api_devs = r.json()
+        assert len(api_devs) == 11
```

Comment:
Why 11? Shouldn't be 10? Is it matching :06:02?

Reply:
Ok, I see it now: it matches a :06:2 mac address. As they represent mac addresses, I was expecting the last octet of the mac to be formatted with 0s, like 01, 02...